### ISSUE TITLE
Problems running MIST with  geth --testnet and problems running geth —testnet —ipcpath /Users/johnwhitton/Library/Ethereum/geth.ipc

### ISSUE THREAD

**Issue by johnwhitton** [ethereum/go-ethereum#2629]
### Problem Summary
I'm unable to open MIST against testnet when running a geth node. Because either
a) if you run `geth --testnet` it doesn't see geth.ipc and tries to start a second node
b) if you run ` geth —testnet —ipcpath /Users/johnwhitton/Library/Ethereum/geth.ipc` it synchs with mainnet

##### Geth version: `geth version`
Johns-MacBook-Pro:~ johnwhitton$ geth version
Geth
Version: 1.4.5-stable-a269a713
Protocol Versions: [63 62 61]
Network Id: 1
Go Version: go1.6.2
OS: darwin
GOPATH=
GOROOT=/usr/local/Cellar/go/1.6.2/libexec
##### OS & Version: OSX
Commit hash : (if `develop`)

#### Expected behaviour
Want to be able to run `geth --testnet` and have MIST connect to the testnet node
_Note : Below is documented with MIST 0.7.4 but same occurs on other machines

#### Actual behaviour
When running `geth -testnet` and opening MIST it tries to open a second node
This is because the geth.ipc file is created under Ethereum/testnet and MIST looks under Ehereum
##### Showing where geth.ipc is created
Johns-MacBook-Pro:Ethereum johnwhitton$ pwd
/Users/johnwhitton/Library/Ethereum
Johns-MacBook-Pro:Ethereum johnwhitton$ ls
chaindata	dapp		keystore	nodekey		nodes		testnet
Johns-MacBook-Pro:Ethereum johnwhitton$ cd testnet/
Johns-MacBook-Pro:testnet johnwhitton$ ls
chaindata	dapp		geth.ipc	nodekey		nodes
Johns-MacBook-Pro:testnet johnwhitton$

##### MIST (Election Log)

Here is the log file showing MIST trying to open a second node
[MIST_Trying_To_Open_A_Second_Node.txt](https://github.com/ethereum/go-ethereum/files/287001/MIST_Trying_To_Open_A_Second_Node.txt)

I believe the key here is _Error: connect ENOENT /Users/johnwhitton/Library/Ethereum/geth.ipc_ once again because geth.ipc is created under Ethereum/testnet (as above)
[MIST_Trying_To_Open_A_Second_Node.txt](https://github.com/ethereum/go-ethereum/files/286970/MIST_Trying_To_Open_A_Second_Node.txt)

### Workaround points to the mainnet instead of testnet
The workaround for this as discussed on gitter with @obscuren is to run geth with an --ipc flag e.g.
`geth —testnet —ipcpath /Users/johnwhitton/Library/Ethereum/geth.ipc`

The problem with this is that for some reason **geth then tries synching to mainnet**
Here is the log file 
[geth_testnet_ipc_issue.txt](https://github.com/ethereum/go-ethereum/files/287008/geth_testnet_ipc_issue.txt)

If you see the attached file when running `geth -testnet` you see it is synching testnet

```I0527 09:42:25.832796 eth/downloader/downloader.go:299] Block synchronisation started
I0527 09:42:28.506958 core/blockchain.go:959] imported 1 block(s) (0 queued 0 ignored) including 2 txs in 838.779764ms. #569313 [7354c47c / 7354c47c]```

However when running `geth —testnet —ipcpath /Users/johnwhitton/Library/Ethereum/geth.ipc` it is synching to mainnet

```I0527 09:44:33.570664 eth/downloader/downloader.go:299] Block synchronisation started
I0527 09:44:51.954394 core/blockchain.go:959] imported 25 block(s) (0 queued 0 ignored) including 0 txs in 758.998931ms. #22722 [0d24e4bd / f64f17c2]```

Also see block 0 returned by `geth attach`

Johns-MacBook-Pro:testnet johnwhitton$ geth attach
instance: Geth/v1.4.5-stable-a269a713/darwin/go1.6.2
coinbase: 0xa1765fa3e1f455f6b2dbd2dc25b02c95d708eab5
at block: 22697 (Sun, 02 Aug 2015 09:27:50 PDT)
 datadir: /Users/johnwhitton/Library/Ethereum
> eth.getBlock(0).hash
"0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3"

#### Steps to reproduce the behaviour
Run geth --testnet
Open MIST


#### Backtrace

````
[backtrace]
````

**Comment by bas-vk** [comment]
Any reason why you use the [EM dash](http://www.fileformat.info/info/unicode/char/2014/index.htm) symbol instead of the ordinary '-' character in your second command where the problem starts?

**Comment by johnwhitton** [comment]
Thanks @bas-vk I'm not sure where the EM dash came from I manually retyped the command and ran it with the following which then connected to testnet and placed geth.ipc in the Ethereum directory
`geth --testnet --ipcpath "/Users/johnwhitton/Library/Ethereum/geth.ipc"`
**So I have a workaround**
If there is another ticket already opened for the initial problem (i.e. MIST trying to start a second node when running `geth --testnet` I suggest linking this ticket to that (or vice-versa) and closing it.

Thanks again for all your help on this.

**Comment by radfish** [comment]
Although unlikely, I wonder if this problem is related to #2634 -- for me, `geth attach` simply ignores its `--ipcpath` argument. So, if you run two nodes, it can get confusing, because `attach` will only attach to the node whose datadir is `$HOME/.ethereum` no matter what you tell it to with `--ipcpath` or `--datadir`. Please ignore if unrelated.

**Comment by bas-vk** [comment]
@johnwhitton, the solution to start geth with the `--testnet` and `--ipcpath` flags is not a workaround but currently the way to go if you want mist to connect on a custom started geth instance on the testnet. The reason is that geth (from version 1.4) will open the default ipc endpoint in the testnet datadir where geth 1.3 would open it in the mainnet datadir. This causes mist not to find the endpoint and trying a new instance of geth.

**Comment by obscuren** [comment]
@radfish try the following: `geth attach ipc:/path/to/geth.ipc`

**Comment by coffeethecup** [comment]
Hi, 
For two days I was playing around with the goal that the geth-client and MIST are using the same Testnet Ethereum chain. 
I tried x times starting the geth before with different --ipcpath definitions. Nothing worked.
The following approach worked:
- No client started before, fresh installation
Installation of geth:
```
sudo apt-get install software-properties-common
sudo add-apt-repository -y ppa:ethereum/ethereum
sudo add-apt-repository -y ppa:ethereum/ethereum-dev
sudo apt-get update
sudo apt-get install ethereum
```
- Start MIST 0.8.4 and end it after some blocks has been synced

- then start for the first time geth with 
`geth` --testnet --ipcpath "/home/youruser/.ethereum/geth.ipc"

Checked several times starting MIST during geth sync, and MIST is on the same chain.
I use UBUNTU 16.4.1 on a Virtualbox image.
